Tests/UIKitTests/UIViewExtensionsTests.swift

```diff
@@ -411,5 +411,21 @@ final class UIViewExtensionsTests: XCTestCase {
 		XCTAssertNotNil(subview.centerYAnchor)
 	}
 
+    func testSuperviewWithClass() {
+        let tableView = UITableView(frame: .zero)
+        let tableViewCell = UITableViewCell(frame: .zero)
+        tableView.addSubview(tableViewCell)
+        let button = UIButton(frame: .zero)
+        tableViewCell.addSubview(button)
+        let buttonSubview = UIView(frame: .zero)
+        button.addSubview(buttonSubview)
+        XCTAssertEqual(button.ancestorView(withClass: UITableViewCell.self), tableViewCell)
+        XCTAssertEqual(button.ancestorView(withClass: UITableView.self), tableView)
+        XCTAssertEqual(button.ancestorView(withClass: UIButton.self), nil)
```

Comment:
XCAssertNil() could be used here :))

Reply:
Sounds good, I will prepare a function with tests for that!